### PR TITLE
New Data Source: 'azurerm_iothub_dps_shared_access_policy'

### PR DESCRIPTION
**Pull request by brunhil**
Add new data source 'azurerm_iothub_dps_shared_access_policy'

=== RUN   TestAccDataSourceAzureRMIotHubDpsSharedAccessPolicy_basic
=== PAUSE TestAccDataSourceAzureRMIotHubDpsSharedAccessPolicy_basic
=== CONT  TestAccDataSourceAzureRMIotHubDpsSharedAccessPolicy_basic
--- PASS: TestAccDataSourceAzureRMIotHubDpsSharedAccessPolicy_basic (303.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub/tests        303.714s

### CODE REVIEW

File: website/docs/d/iothub_dps_shared_access_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+* `id` - The ID of the IoT Hub Device Provisioning Service Shared Access Policy.
+
+* `primary_key` - The primary key used to create the authentication token.
+
+* `primary_connection_string` - The primary connection string of the Shared Access Policy.
+
+* `secondary_key` - The secondary key used to create the authentication token.
+
+* `secondary_connection_string` - The secondary connection string of the Shared Access Policy.
```
Comment (katbyte) [inline website/docs/d/iothub_dps_shared_access_policy.html.markdown]: Could we make the language here consistent with the rest of the data source documentation?

File: azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			return fmt.Errorf("Error: IotHub DPS %q (Resource Group %q) was not found", iothubDpsName, resourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving IotHub DPS %q (Resource Group %q): %+v", iothubDpsName, resourceGroup, err)
+	}
+
+	accessPolicy, err := client.ListKeysForKeyName(ctx, iothubDpsName, keyName, resourceGroup)
+	if err != nil {
+		if utils.ResponseWasNotFound(accessPolicy.Response) {
+			return fmt.Errorf("Error: IotHub DPS Shared Access Policy %q (Resource Group %q) was not found", iothubDpsName, resourceGroup)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go]: we should output the key name here too:

```suggestion
			return fmt.Errorf("Error: Shared Access Policy %q (IotHub DPS %q / Resource Group %q) was not found", keyName, iothubDpsName, resourceGroup)
```

File: azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		}
+
+		return fmt.Errorf("Error loading IotHub DPS Shared Access Policy %q (IotHub DPS %q / Resource Group %q): %+v", keyName, iothubDpsName, resourceGroup, err)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go]:
```suggestion
		return fmt.Errorf("Error loading Shared Access Policy %q (IotHub DPS %q / Resource Group %q): %+v", keyName, iothubDpsName, resourceGroup, err)
```

File: azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	d.Set("name", keyName)
+	d.Set("resource_group_name", resourceGroup)
+
+	resourceID := fmt.Sprintf("%s/keys/%s", *iothubDps.ID, keyName)
+	d.SetId(resourceID)
+
+	d.Set("primary_key", accessPolicy.PrimaryKey)
+	d.Set("secondary_key", accessPolicy.SecondaryKey)
+
+	if props := iothubDps.Properties; props != nil {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go]: could we ensure the fields `primary_connection_string` and `secondary_connection_string` are always set here?

Comment (tombuildsstuff) [reply azurerm/internal/services/iothub/data_source_iothub_dps_shared_access_policy.go]: update: just noticed this is the same in the Resource too - I'll push a commit to fix this

**Comment by tombuildsstuff** [comment]
Test passes:

Screenshot 2020-02-11 at 16 25 19

**Pull request merged**